### PR TITLE
update stencil version to support stencil ^3

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

> Upgrades version of @stencil/core to ^3
> Fixes issue where using `@stencil/core` higher than version 2 resulted in a number of duplicate identifier errors

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

- Jira issue number: (PUT IT HERE)
- Process.st launch checklist: (PUT IT HERE)

### Development

- [ ] [Prettier](https://www.npmjs.com/package/prettier) was run (if applicable)
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist

### Code review

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

**Comment by 00salmon** [comment]
I imagine this issue will come up again shortly now that `@stencil/core` has a 4.0 release, so we may want to also release a 4.0.0 version as well. there's probably some package.json magic we can add to enforce these version requirements better

**Comment by 00salmon** [comment]
I imagine this issue will come up again shortly now that `@stencil/core` has a 4.0 release, so we may want to also release a 4.0.0 version as well. there's probably some package.json magic to enforce these version requirements better

**Comment by ayangweb** [comment]
> I imagine this issue will come up again shortly now that `@stencil/core` has a 4.0 release, so we may want to also release a 4.0.0 version as well. there's probably some package.json magic we can add to enforce these version requirements better

I think so, because `@stencil/core` creates projects with version 4.0.0 by default now!

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,9 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [3.0.0] - 2023-07-10
```
Comment (00salmon) [inline CHANGELOG.md]:
```suggestion
## [3.0.0] - 2023-07-27
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@saasquatch/stencil-hooks",
-  "version": "2.0.3",
+  "version": "3.0.0-1",
```
Comment (00salmon) [inline package.json]:
```suggestion
  "version": "3.0.0",
```

**Review by Locrian24 — DISMISSED** [review]

**Review by Locrian24 — DISMISSED** [review]

**Review by Locrian24 — APPROVED** [review]

**Pull request merged**